core-bundle/src/Resources/contao/library/Contao/InsertTags.php

```diff
@@ -898,9 +898,9 @@ protected function doReplace($strBuffer, $blnCache)
 							$objFile = new File(rawurldecode($src));
 
 							// Add the image dimensions
-							if (($imgSize = $objFile->imageSize) !== false)
+							if (isset($objFile->imageSize[0], $objFile->imageSize[1]))
 							{
-								$dimensions = ' width="' . StringUtil::specialchars($imgSize[0]) . '" height="' . StringUtil::specialchars($imgSize[1]) . '"';
+								$dimensions = ' width="' . StringUtil::specialchars($objFile->imageSize[0]) . '" height="' . StringUtil::specialchars($objFile->imageSize[1]) . '"';
```

Comment:
@ausi Why are we using `StringUtil::specialchars()` here? Would an `(int)` cast not make more sense?

Reply:
Both fine with me.

Reply:
The HTML spec says, `width` and `height` must contain "valid non-negative integers" ([spec](https://html.spec.whatwg.org/multipage/embedded-content-other.html#attr-dim-width)). So I think a cast would be fine here.

Reply:
Actually, we do not have to cast at all, because `$objFile->imageSize[0]` and `$objFile->imageSize[1]` are always integers. They are set from `ImageDimensions->getSize()->getWidth()` and `->getHeight()` and we use them without a cast or specialchars() in other places, too. Therefore dbd24bda659f1e9aed3190683aa1d2d289fe43c8.